src/syft/core/garbage_collection/gc_batched.py

```diff
@@ -0,0 +1,99 @@
+"""A simple garbage collection heuritics."""
+# stdlib
+from collections import defaultdict
+from typing import Dict
+from typing import List
+
+# syft relative
+from ..common.uid import UID
+from ..node.common.action.garbage_collect_batched_action import (
+    GarbageCollectBatchedAction,
+)
+from ..node.common.client import Client
+from ..pointer.pointer import Pointer
+from .gc_strategy import GCStrategy
+
+
+class GCBatched(GCStrategy):
+    """The GCBatched Strategy."""
+
+    __slots__ = [
+        "client_to_obj_ids",
+        "threshold_client",
+        "threshold_total",
+        "count_total",
+    ]
+
+    client_to_obj_ids: Dict[Client, List[UID]]
+    threshold_client: int
+    threshold_total: int
+    count_total: int
+
+    def __init__(self, threshold_client: int = 10, threshold_total: int = 50) -> None:
+        """Construct the GCBatched Strategy.
+
+        Args:
+            threshold_client (int): the threshold after which a message
+                would be sent to a client to delete all the objects
+                that were recorded for them
+            threshold_total (int): the threshold after which a message
+                would be sent to all clients to delete all the objects
+                that were recorded
+
+        Return:
+            None
+        """
+        self.client_to_obj_ids = defaultdict(list)
+        self.threshold_client = threshold_client
+        self.threshold_total = threshold_total
```

Comment:
Do we need both in your opinion?

Reply:
I think we might need the ```threshold_total``` in case the DS is talking with a lot of DOs (and it might accumulate a lot of pointers? even if they don't take as much space as the real data, they might also add up) - and in case we never reach the limit for ```trehshold_client``` we might consume memory on the DS side?

Thinking also about this now...I think I would remove this, this is premature optimization and we should add code only in case we need it (for the moment there is no indication that this might happen).